### PR TITLE
Support CPU resource configurable for Kubernates job under MoK Mode

### LINKED ISSUE #16007: Make CPU config of Container configurable under MoK mode
### Description

In the Druid cluster with MoK mode, we can observe that the CPU usage of task Pod always touched the 100% limit, which affecting the performance of ingestion progress.

### Motivation

After researched the relevant resource code, we can discovered that the CPU configuration for Container in K8s cluster has been fixed as `new Quantity("1000", "m")`, which is equal to one core. It would be better to make it configurable to handle various business scenarios.

### PR DESCRIPTION
**Pull request by fectrain**
Fixes #16007 .

### Description
Make CPU core configurable for Kubernetes job(Peon) under MoK Mode

Currently the CPU core for Peon is fixed to 1, this PR is to support adjusting it by adding a Overlord property: `druid.indexer.runner.cpuCoreInMicro=2000`

**Verification**

截屏2024-02-29 下午5 42 59

image

#### Release note

<hr>

##### Key changed/added classes in this PR
 * `KubernetesTaskRunnerConfig`
 * `PeonCommandContext`
 * `K8sTaskAdapter`
 * `DruidK8sConstants`

<hr>

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [x] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by georgew5656** [comment]
will take a look at this today, looks pretty reasonable

File: extensions-contrib/kubernetes-overlord-extensions/src/test/java/org/apache/druid/k8s/overlord/taskadapter/K8sTaskAdapterTest.java [hunk 1]
```diff
@@ -572,14 +576,72 @@ void testEphemeralStorageIsRespected() throws IOException
     Assertions.assertEquals(expected, actual);
   }
 
+  @Test
+  void testEphemeralResourceIsEspected() throws IOException
```
Comment (georgew5656) [inline extensions-contrib/kubernetes-overlord-extensions/src/test/java/org/apache/druid/k8s/overlord/taskadapter/K8sTaskAdapterTest.java:580]: i think this test name is wrong, seems like it should be more like testCPUResourceIsRespected

File: docs/development/extensions-contrib/k8s-jobs.md [hunk 2]
```diff
@@ -219,23 +219,24 @@ data:
...
-|`druid.indexer.runner.peonMonitors`| `JsonArray`     | Overrides `druid.monitoring.monitors`. Use this property if you don't want to inherit monitors from the Overlord.                                                                                                                                |`[]`|No|
-|`druid.indexer.runner.graceTerminationPeriodSeconds`| `Long`          | Number of seconds you want to wait after a sigterm for container lifecycle hooks to complete.  Keep at a smaller value if you want tasks to hold locks for shorter periods.                                                                      |`PT30S` (K8s default)|No|
-|`druid.indexer.runner.capacity`| `Integer`       | Number of concurrent jobs that can be sent to Kubernetes.                                                                                                                                                                                        |`2147483647`|No|
```
Comment (georgew5656) [inline docs/development/extensions-contrib/k8s-jobs.md:238]: can you use the intellij props from here so the whole table doesn't get replaced like this? https://github.com/apache/druid/blob/master/dev/druid_intellij_formatting.xml#L77-L80

intellij config instructions: https://github.com/apache/druid/blob/master/dev/intellij-setup.md

**Review by georgew5656 — COMMENTED** [review]
logic looks good to me, left a couple small comments on docs and changing the name of a test

File: docs/development/extensions-contrib/k8s-jobs.md (hunk 2 above)
Comment (fectrain) [reply docs/development/extensions-contrib/k8s-jobs.md:238]: done.

File: extensions-contrib/kubernetes-overlord-extensions/src/test/java/org/apache/druid/k8s/overlord/taskadapter/K8sTaskAdapterTest.java (hunk 1 above)
Comment (fectrain) [reply extensions-contrib/kubernetes-overlord-extensions/src/test/java/org/apache/druid/k8s/overlord/taskadapter/K8sTaskAdapterTest.java]: updated, thanks!

**Review by georgew5656 — APPROVED** [review]

**Pull request merged**